channeldb/invoices.go

```diff
@@ -223,6 +242,9 @@ type Invoice struct {
 	// NOTE: This index starts at 1.
 	SettleIndex uint64
 
+	// State describes the state the invoice is in.
+	State ContractState
```

Comment:
If the `ContractTerm` field is staying, then why move this _outside_ of it?

Reply:
see the other inline comment re: `ContractTerm`, but the main rationale is that `State` is more metadata than it is a constraint on the sender.